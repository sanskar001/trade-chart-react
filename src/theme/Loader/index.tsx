import React from "react";

interface LoaderProps {
  style?: React.CSSProperties;
}

const Loader: React.FC<LoaderProps> = ({ style }) => {
  return (
    <div className="h-full flex items-center justify-center">
      <div className="loader" style={style}></div>
    </div>
  );
};

export default Loader;
