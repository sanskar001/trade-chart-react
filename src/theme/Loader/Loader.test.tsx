import renderer from "react-test-renderer";
import Loader from ".";

it("renders correctly when there is default loader", () => {
  const tree = renderer.create(<Loader />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly when there is loader with custom style", () => {
  const tree = renderer
    .create(
      <Loader style={{ width: "64px", height: "64px", color: "orange" }} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
