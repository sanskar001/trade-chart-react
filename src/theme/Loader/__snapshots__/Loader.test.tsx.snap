// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there is default loader 1`] = `
<div
  className="h-full flex items-center justify-center"
>
  <div
    className="loader"
  />
</div>
`;

exports[`renders correctly when there is loader with custom style 1`] = `
<div
  className="h-full flex items-center justify-center"
>
  <div
    className="loader"
    style={
      Object {
        "color": "orange",
        "height": "64px",
        "width": "64px",
      }
    }
  />
</div>
`;
